ycsb-mongodb/mongodb/src/main/java/com/yahoo/ycsb/db/MongoDbClient.java

```diff
@@ -233,6 +344,47 @@ private static AutoEncryptionSettings generateEncryptionSettings(String url, Boo
         return autoEncryptionSettings;
     }
 
+    private static ArrayList<Long> parseCommaSeparatedIntegers(String toParse, int outputListSize, long defaultValue) {
+        if (toParse.trim().isEmpty()) {
+            return null;
+        }
+        String[] values = toParse.split(",", -1);
+
+        if (outputListSize < 0) {
+            outputListSize = values.length;
+        }
+        ArrayList<Long> parsedValues = new ArrayList<Long>(Collections.nCopies(outputListSize, defaultValue));
+        int limit = Math.min(values.length, outputListSize);
+        for (int i = 0; i < limit; i++) {
+            String v = values[i].trim();
+            parsedValues.set(i, v.isEmpty() ? defaultValue : Long.parseLong(v));
+        }
+        return parsedValues;
+    }
+
+    private static HashMap<String, DiscreteGenerator> createDiscreteFieldsMap(String cardinalities) {
+        ArrayList<Long> parsedCardinalities = parseCommaSeparatedIntegers(cardinalities, -1, 0);
+        if (parsedCardinalities == null || parsedCardinalities.isEmpty()) {
+            return null;
+        }
+        HashMap<String, DiscreteGenerator> outputMap = new HashMap<String, DiscreteGenerator>();
+        for (int i = 0; i < parsedCardinalities.size(); i++) {
+            Long value = parsedCardinalities.get(i);
+            if (value <= 0) {
+                continue;
+            }
+            DiscreteGenerator gen = new DiscreteGenerator();
+            for (long j = 0; j < value; j++) {
+                gen.addValue(1, "value" + j);
+            }
+            outputMap.put("field" + i, gen);
+        }
+        if (outputMap.isEmpty()) {
```

Comment:
you could drop this check, functionally there is little difference between empty and null.

Reply:
done